dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsConnector.java

```diff
@@ -17,433 +17,88 @@
-        "L.StatementType",
-        "L.AppID",
-        "L.DefaultDatabase",
-        "L.ErrorCode",
-        "L.ErrorText",
-        "L.FirstRespTime",
-        "L.LastRespTime",
-        "L.NumResultRows",
-        "L.QueryText",
-        "L.ReqPhysIO",
-        "L.ReqPhysIOKB",
-        "L.RequestMode",
-        "L.SessionID",
-        "L.SessionWDID",
-        "L.Statements",
-        "L.TotalIOCount",
-        "L.WarningOnly",
-        "L.StartTime"
-    };
-
-    public static boolean isQueryTable(@Nonnull String expression) {
-        return expression.startsWith("ST.");
-    }
-
-    public TeradataLogsConnector() {
-        super("teradata-logs");
-    }
-
-    // to proxy for Terdata14LogsConnector
-    protected TeradataLogsConnector(@Nonnull String name) {
-        super(name);
-    }
-
-    /** This is shared between all instances of TeradataLogsJdbcTask. */
-    protected static class SharedState {
-
-        /**
-         * Whether a particular expression is valid against the particular target Teradata version.
-         * This is a concurrent Map of immutable objects, so is threadsafe overall.
-         */
-        private final ConcurrentMap<String, Boolean> expressionValidity = new ConcurrentHashMap<>();
+public class TeradataLogsConnector extends AbstractTeradataConnector
+    implements LogsConnector, TeradataLogsDumpFormat {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TeradataLogsConnector.class);
+
+  public TeradataLogsConnector() {
+    super("teradata-logs");
+  }
+
+  // to proxy for Terdata14LogsConnector
+  protected TeradataLogsConnector(@Nonnull String name) {
+    super(name);
+  }
+
+  @Override
+  public void addTasksTo(List<? super Task<?>> out, @Nonnull ConnectorArguments arguments)
+      throws MetadataDumperUsageException {
+    out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
+    out.add(new FormatTask(FORMAT_NAME));
+
+    TeradataSqlQueryFactoryBuilder builder =
+        TeradataSqlQueryFactoryBuilder.startBuildingFrom(arguments);
+    Class<? extends Enum<?>> headerClass =
+        arguments.isAssessment() ? HeaderForAssessment.class : Header.class;
+
+    // Beware of Teradata SQLSTATE HY000. See issue #4126.
+    // Most likely caused by some operation (equality?) being performed on a datum which is too long
+    // for a varchar.
+    ZonedIntervalIterable intervals = ZonedIntervalIterable.forConnectorArguments(arguments);
+    LOG.info("Exporting query log for " + intervals);
+    SharedState state = new SharedState();
+    for (ZonedInterval interval : intervals) {
+      String file =
+          ZIP_ENTRY_PREFIX
+              + DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(interval.getStartUTC())
+              + ".csv";
+      SqlQueryFactory factory = builder.within(interval).build();
+      out.add(new TeradataLogsJdbcTask(file, state, factory).withHeaderClass(headerClass));
     }
+  }
 
-    protected static class TeradataLogsJdbcTask extends AbstractJdbcTask<Void> {
```

Comment:
I would suggest to extract the `TeradataLogsJdbcTask` to a separate file in one PR and do other changes in separate PRs.